compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java

```diff
@@ -1134,14 +1139,25 @@ public void visit(BLangRecordLiteral recordLiteral) {
                     analyzeNode(keyValuePair.key.expr, env);
                 }
                 analyzeNode(keyValuePair.valueExpr, env);
+                addToMappingConstructorExprs(keyValuePair);
             } else if (field.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
                 analyzeNode((BLangRecordLiteral.BLangRecordVarNameField) field, env);
+                addToMappingConstructorExprs(field);
             } else {
-                analyzeNode(((BLangRecordLiteral.BLangRecordSpreadOperatorField) field).expr, env);
+                analyzeNode((BLangRecordLiteral.BLangRecordSpreadOperatorField) field, env);
             }
         }
     }
 
+    @Override
+    public void visit(BLangRecordLiteral.BLangRecordSpreadOperatorField spreadOperatorField) {
+        analyzeNode(spreadOperatorField.expr, env);
+        NodeKind kind = spreadOperatorField.expr.getKind();
+        if (!(kind == NodeKind.RECORD_LITERAL_EXPR || kind == NodeKind.TYPE_CONVERSION_EXPR)) {
```

Comment:
What is this special case we're trying to handle here? Why do we need to special case the type cast expression?

Reply:
In here we only need to add whats inside the mapping constructor (key values and simple var refs).

Reply:
Inside the spread operator field, the expression can be type cast expression.

Reply:
Spread (with a mapping constructor) can be other expressions too, not just these?

Reply:
Only these will be allowed to pass as a function argument. Others will be not allowed by the type checking phase.

Reply:
Is that statement correct? 

```ballerina
type Record record {|
    string[] arr1;
    string[] arr2;
    string[] arr3;
|};

function qux() {
    f(...<Record> {arr1: [], arr2: [], ...({arr3: []})}); // grouped expression in spread
}

isolated function f(string[] arr1, string[] arr2, string[] arr3) {
}
```

Reply:
I didn't figure out this is possible previously. Handle it in [commit](https://github.com/ballerina-platform/ballerina-lang/pull/41772/commits/d7e77290ba3bfc7bf08e9e6703fee8cc132aeea9).